main.py

```diff
@@ -313,6 +317,117 @@ async def get_running_experiment() -> Optional[int]:
+    dst_result_path = posixpath.join(rid_dir_path, "result.h5")
+    shutil.copyfile(src_result_path, dst_result_path)
+    # modify the result file
+    with h5py.File(dst_result_path, "a") as result_file:
+        # start time
+        start_time = result_file["start_time"][()]
+        del result_file["start_time"]
+        start_time_str = datetime.fromtimestamp(start_time).isoformat()
+        start_time_dataset = result_file.create_dataset(
+            "start_time", (1,), dtype=h5py.string_dtype(encoding="utf-8")
+        )
+        start_time_dataset[0] = start_time_str
+        # run time
+        run_time = result_file["run_time"][()]
+        del result_file["run_time"]
+        run_time_str = datetime.fromtimestamp(run_time).isoformat()
+        run_time_dataset = result_file.create_dataset(
+            "run_time", (1,), dtype=h5py.string_dtype(encoding="utf-8")
+        )
+        run_time_dataset[0] = run_time_str
+        # submission time
+        submission_time_dataset = result_file.create_dataset(
+            "submission_time", (1,), dtype=h5py.string_dtype(encoding="utf-8")
+        )
+        submission_time_dataset[0] = submission_time_str
+        # visualize option
+        visualize_dataset = result_file.create_dataset("visualize", (1,), dtype="bool")
+        visualize_dataset[0] = visualize
+    # move the modified experiment file to the RID directory
+    if visualize:
+        src_experiment_path = posixpath.join(
+            result_dir_path, f"experiment_{submission_time_str}.py"
+        )
+        dst_experiment_path = posixpath.join(rid_dir_path, "modified_experiment.py")
+        shutil.move(src_experiment_path, dst_experiment_path)
+    # remove the metadata file
+    os.remove(metadata_path)
+    return True
+
+
+@app.get("/result/")
+async def list_result_directory() -> List[int]:
+    """Post-processes the executed experiments and returns the RID list.
+
+    It performs the following:
+        1. Organize the result directory.
+        2. Return the RID list.
+
+    Returns:
+        A list with RIDs of the executed experiments, sorted in an ascending order.
+    """
+    result_dir_path = posixpath.join(configs["master_path"], configs["result_path"])
+    # navigate to each RID directory not organized yet
+    for item in os.listdir(result_dir_path):
+        item_path = posixpath.join(result_dir_path, item)
+        if posixpath.isdir(item_path) and item.startswith("_") and item[1:].isdigit():
+            rid = item[1:]
+            if organize_result_directory(result_dir_path, rid):
+                rid_dir_path = posixpath.join(result_dir_path, f"{rid}/")
+                os.rename(item_path, rid_dir_path)
+    # find all organized RID directories
+    rid_list = []
+    for item in os.listdir(result_dir_path):
+        item_path = posixpath.join(result_dir_path, item)
+        if posixpath.isdir(item_path) and item.isdigit():
+            rid_list.append(int(item))
+    rid_list.sort()
```

Comment:
So this sort means:
"Return all experiments that have been completed, in ascending order of rid."
Am I right?

Reply:
Right! Is there any problem?

Reply:
Not at all.